src/org/stringtemplate/v4/STGroup.java

```diff
@@ -80,9 +80,11 @@
 		TEMPLATE_FILE_EXTENSION = ".st";
 	}
 
-    /** When we use key as a value in a dictionary, this is how we signify. */
-    public static final String DICT_KEY = "key";
-    public static final String DEFAULT_KEY = "default";
+    /** When we use key as a value in a dictionary, this is how we signify.
+     *  Magic suffix is used to distinguish "default" string literal and "default" key
+     *  End part is the date of this suffix adding. */
+    public static final String DICT_KEY = "key_st4_magic_suffix_2016/01/10";
```

Comment:
I would actually use this:

```java
public static final String DICT_KEY = new String("key");
```

This guarantees that the string `DICT_KEY` is different from any other string running in the application, so when it gets compared by reference there is no problem.